packages/liveblocks-react/src/factory.tsx

```diff
@@ -617,50 +618,48 @@ export function createRoomContext<
     key: TKey
   ): TStorage[TKey] | null {
     const room = useRoom();
-    const root = useMutableStorageRoot();
+    const rootOrNull = useMutableStorageRoot();
     const rerender = useRerender();
 
     React.useEffect(() => {
-      if (root === null) {
+      if (rootOrNull === null) {
         return;
       }
+      const root = rootOrNull;
 
-      let liveValue = root.get(key);
+      let unsubCurr: (() => void) | undefined;
+      let curr = root.get(key);
+
+      function subscribeToCurr() {
+        unsubCurr = isLiveNode(curr)
+          ? room.subscribe(curr, rerender)
+          : undefined;
+      }
 
       function onRootChange() {
-        const newCrdt = root!.get(key);
-        if (newCrdt !== liveValue) {
-          unsubscribeCrdt();
-          liveValue = newCrdt;
-          unsubscribeCrdt = room.subscribe(
-            liveValue as any /* AbstractCrdt */, // TODO: This is hiding a bug! If `liveValue` happens to be the string `"event"` this actually subscribes an event handler!
-            rerender
-          );
+        const newValue = root.get(key);
+        if (newValue !== curr) {
+          unsubCurr?.();
+          curr = newValue;
+          subscribeToCurr();
           rerender();
         }
       }
 
-      let unsubscribeCrdt = room.subscribe(
-        liveValue as any /* AbstractCrdt */, // TODO: This is hiding a bug! If `liveValue` happens to be the string `"event"` this actually subscribes an event handler!
-        rerender
-      );
-      const unsubscribeRoot = room.subscribe(
-        root as any /* AbstractCrdt */, // TODO: This is hiding a bug! If `liveValue` happens to be the string `"event"` this actually subscribes an event handler!
```

Comment:
This also fixes this (theoretic) bug we already knew about. In a way, it's a special case of the same bug. No longer!